packages/aws-cdk/lib/cli/telemetry/file-sink.ts

```diff
@@ -38,23 +37,19 @@ export class FileTelemetrySink implements ITelemetrySink {
       throw new ToolkitError(`Telemetry file already exists at ${this.logFilePath}`);
     }
 
-    // Create the file if necessary
-    const directory = path.dirname(this.logFilePath);
-    if (!fs.existsSync(directory)) {
-      fs.mkdirSync(directory, { recursive: true });
-    }
+    // Create the file
+    fs.ensureFileSync(this.logFilePath);
+    fs.writeJsonSync(this.logFilePath, []);
   }
 
   /**
    * Emit an event.
    */
   public async emit(event: TelemetrySchema): Promise<void> {
     try {
-      // Format the events as a JSON string with pretty printing
-      const output = JSON.stringify(event, null, 2) + '\n';
-
-      // Write to file
-      fs.appendFileSync(this.logFilePath, output);
+      const json = fs.readJsonSync(this.logFilePath);
+      json.push(event);
+      fs.writeJSONSync(this.logFilePath, json, { spaces: 2 });
```

Comment:
I originally used `appendFileSync()` because it's ~50% faster than read + write. However we are not emitting that many telemetry events at the moment and I think this is fine unless we add many many more events. The payoff is that we can write valid json and the telemetry file can be `.json`.